andela/ah-scorpion
# #159053982 Send verification email to user on registration

**What does this PR do?**
Enables the system to send an account verification email to the new user upon registration, for the user to activate their account

#### Description of Task to be completed?
 - added `account_activator.py` to activate account when user clicks verification link'
 - added an HTML template for the verification email to be sent to the user
 - modified the `User` model to set the default value for `is_active` from `True` to `False`
 - added `id` and `is_active` to be returned by the serializer
 - added activate account endpoint to `urls.py` to be used when the user clicks verification email
 - added email sender lines to `views.py` to send a verification token to user
 - added sender email settings to `settings.py` to be read from the `.env` file
 - added email settings in the `.env` file

#### How should this be manually tested?
 - use git pull to get the source code from `ft-email-verification-159053982` branch
 - update your `.env` file to include your email settings
 - export the `.env` using `source .env`
 - start the server using `python manage.py runserver`
 - using `Postman` make a post request in the following format (Replace values inside the <> with your own values):
```
{
"user":{
		"email":"<your email address>",
		"username":"<your username>",
		"password":"<your password>"
        }
}
```
 - you will receive an email in the email address specified above.
 - click on the verification link on the email and you should receive the following message on your browser `Thank you for confirming your email address. Welcome to Authors' Haven.` The user account will now be active.

#### What are the relevant pivotal tracker stories?
 - [#159053982](https://www.pivotaltracker.com/story/show/159053982)
